### ISSUE TITLE
Debate: absolute links

### ISSUE THREAD

**Issue by vladmunteanu** [vladmunteanu/starlette-jsonapi#14]
In some scenarios, we might need to generate absolute links. Ideally this should be configurable, but it requires some refactoring of `JSONAPISchema` and `JSONAPIRelationship` in order to use `request` instead of `app` when generating absolute links.

Pros:
- an absolute URL is likely easier to use for a client

Cons:
- it complicates some tests, since it is trickier to create a `request` than an `app`
- we might need to preserve query parameters in _some_ cases (ie: top level links vs object level links when considering pagination), and this might get too complex.

**Comment by khornberg** [comment]
I think the domain part of the url can be provided when instantiating the app

Modifying `generate_url` with
```
    def generate_url(self, link, **kwargs):
        if self.app and isinstance(self.app, Starlette) and link:
            prefix = getattr(self.app, "url_prefix", "")
            return f"{prefix}{self.app.url_path_for(link, **kwargs)}"
        return None

```

and instantiating the app with
```
    app.url_prefix = 'https://example.com'

```

Using the example postman collection (which is a great resource) to create an organization

I get 
```
{
    "data": {
        "type": "organizations",
        "id": "1",
        "attributes": {
            "name": "some-name",
            "contact_phone": "some-contact-phone",
            "contact_url": "some-contact-url"
        },
        "links": {
            "self": "https://example.com/api/organizations/1"
        }
    },
    "links": {
        "self": "https://example.com/api/organizations/1"
    }
}
```

This allows the url to remain relative or be absolute
this allows tests to be consistent
this allows various implementations of how to set the url e.g through the environment, a config file, some network call, etc.

**Comment by vladmunteanu** [comment]
That's a very good point, @khornberg!
I think I like this way better than handling it ourselves, especially because query parameters are tough to consider from all various angles.

**Comment by vladmunteanu** [comment]
We should probably add this to the Readme too. I can do it during the weekend.

**Comment by vladmunteanu** [comment]
Implemented in #16
